corehq/apps/users/tests/test_models.py

```diff
@@ -94,3 +95,117 @@ def setUp(self):
 
     def _get_domain_by_name(self, name):
         return self.domains[name]
+
+
+class DeviceAppMetaMergeTests(SimpleTestCase):
+    def test_overwrites_key(self):
+        self.original_meta.num_unsent_forms = 5
+        self.updates_meta.num_unsent_forms = 2
+
+        self.original_meta.merge(self.updates_meta)
+        self.assertEqual(self.original_meta.num_unsent_forms, 2)
+
+    def test_ignores_older_submissions(self):
+        recent_meta = DeviceAppMeta(num_unsent_forms=5, last_request=self.current_time)
+        older_meta = DeviceAppMeta(num_unsent_forms=2, last_request=self.previous_time)
+
+        recent_meta.merge(older_meta)
+        self.assertEqual(recent_meta.num_unsent_forms, 5)
+
+    def test_ignores_simultaneous_submissions(self):
+        meta1 = DeviceAppMeta(num_unsent_forms=5, last_request=self.current_time)
+        meta2 = DeviceAppMeta(num_unsent_forms=2, last_request=self.current_time)
+
+        meta1.merge(meta2)
+        self.assertEqual(meta1.num_unsent_forms, 5)
+
+    def test_merges_new_properties(self):
+        self.updates_meta.num_unsent_forms = 5
+
+        self.original_meta.merge(self.updates_meta)
+        self.assertEqual(self.original_meta.num_unsent_forms, 5)
+
+    def test_merges_new_dates(self):
+        self.updates_meta.last_heartbeat = self.current_time
+
+        self.original_meta.merge(self.updates_meta)
+        self.assertEqual(self.original_meta.last_heartbeat, self.current_time)
+
+    def test_does_not_overwrite_unspecified_properties(self):
```

Comment:
Did you mean to swap `updates_meta` and `original_meta` in this test?  Judging from the method name, I'd expect `original_meta` to be the one with an extra property.  That and this body is identical to `test_merges_new_properties`.

Reply:
Thanks, that was an oversight for sure